File: src/main/java/destiny/astrology/eclipse/LunarEclipseTotalObservation.java
/**
 * Created by smallufo on 2017-11-18.
 */
package destiny.astrology.eclipse;

import destiny.astrology.Azimuth;

public class LunarEclipseTotalObservation extends LunarEclipsePartialObservation {

  private final boolean totalBeginVisible;
  private final boolean totalEndVisible;

  public LunarEclipseTotalObservation(double gmtJulDay, double lng, double lat, double alt, Azimuth azimuth, AbstractLunarEclipse.LunarType lunarType, double magUmbra, double magPenumbra, boolean maxVisible, boolean penumbraBeginVisible, boolean penumbraEndVisible, boolean partialBeginVisible, boolean partialEndVisible, boolean totalBeginVisible, boolean totalEndVisible) {
    super(gmtJulDay, lng, lat, alt, azimuth, lunarType, magUmbra, magPenumbra, maxVisible, penumbraBeginVisible, penumbraEndVisible, partialBeginVisible, partialEndVisible);
    this.totalBeginVisible = totalBeginVisible;
    this.totalEndVisible = totalEndVisible;
  }

  public boolean isTotalBeginVisible() {
    return totalBeginVisible;
  }

  public boolean isTotalEndVisible() {
    return totalEndVisible;
  }
}

File: src/main/java/destiny/astrology/eclipse/LunarEclipseTotalObservation.kt
/**
 * Created by smallufo on 2017-11-18.
 */
package destiny.astrology.eclipse

import destiny.astrology.Azimuth

/**
 * 月全食
 * [ILunarEclipseTotalObservation]
 * */
class LunarEclipseTotalObservation(gmtJulDay: Double, lng: Double, lat: Double, alt: Double, azimuth: Azimuth,
                                   lunarType: ILunarEclipse.LunarType,
                                   magUmbra: Double,
                                   magPenumbra: Double,
                                   maxVisible: Boolean,
                                   penumbraBeginVisible: Boolean,
                                   penumbraEndVisible: Boolean,
                                   partialBeginVisible: Boolean,
                                   partialEndVisible: Boolean,
                                   val totalBeginVisible: Boolean,
                                   val totalEndVisible: Boolean) :
  LunarEclipsePartialObservation(gmtJulDay, lng, lat, alt, azimuth, lunarType,
    magUmbra, magPenumbra, maxVisible, penumbraBeginVisible, penumbraEndVisible, partialBeginVisible, partialEndVisible)
